apps/accounts/signal_handlers.py

```diff
@@ -23,6 +23,9 @@
 
 @receiver(pre_save, sender=Account)
 def on_account_pre_save(sender, instance, **kwargs):
+    if getattr(instance, 'skip_history_when_saving', False):
+        return
+
     if instance.version == 0:
         instance.version = 1
     else:
```

Comment:
Based solely on the provided snippet of code, there doesn't seem to be any obvious issues that could lead to irregularities. However, without the entire context of the implementation (i.e., whether this function is part of a larger framework or Django project), we can't fully evaluate its performance and efficiency implications.

For future reference, it'd be helpful to include more complete lines from the code so I can assess any additional complexities like imports/exports, loops/conditionals/cases nesting, etc.

Additionally, please also ensure all variables are properly annotated with comments where applicable.